lib/src/main/java/com/dynatrace/metric/util/DynatraceMetadataEnricher.java

```diff
@@ -0,0 +1,192 @@
+   *     sign on each line, respectively. If no line is parsable, returns an empty list. Dimensions
+   *     are not normalized.
+   */
+  static List<Dimension> parseDynatraceMetadata(Collection<String> lines) {
+    ArrayList<Dimension> entries = new ArrayList<>();
+
+    // iterate all lines from metadata file.
+    for (String line : lines) {
+      logger.info(String.format("parsing Dynatrace metadata: %s", line));
+      // if there are more than one '=' in the line, split only at the first one.
+      String[] split = line.split("=", 2);
+
+      // skip if there is no '=' in the line
+      if (split.length != 2) {
+        logger.warning(String.format("could not parse metadata line ('%s')", line));
+        continue;
+      }
+
+      String key = split[0];
+      String value = split[1];
+
+      // make sure key and value are set to non-null, non-empty values
+      if ((key == null || key.isEmpty()) || (value == null || value.isEmpty())) {
+        logger.warning(String.format("could not parse metadata line ('%s')", line));
+        continue;
+      }
+      entries.add(Dimension.create(key, value));
+    }
+    return entries;
+  }
+
+  /**
+   * Get the file name of the file in which the metadata is stored.
+   *
+   * @param fileContents A {@link Reader} object containing the contents of the indirection file.
+   * @return The string containing the filename or null if the file is empty.
+   * @throws IOException if an error occurs during reading of the file.
+   */
+  static String getMetadataFileName(Reader fileContents) throws IOException {
+    if (fileContents == null) {
+      throw new IOException("passed Reader cannot be null.");
+    }
+
+    String oneAgentMetadataFileName = null;
+
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
+      String line = reader.readLine();
+      if (line != null && !line.isEmpty()) {
+        oneAgentMetadataFileName = line;
+      }
+    }
+    return oneAgentMetadataFileName;
+  }
+
+  /**
+   * Read the actual content of the metadata file.
+   *
+   * @param fileContents A {@link Reader} object containing the metadata contents.
+   * @return A {@link List<String>} containing the {@link String#trim() trimmed} lines.
+   * @throws IOException if an error occurs during reading of the file.
+   */
+  static List<String> getDynatraceMetadataFileContents(Reader fileContents) throws IOException {
+    if (fileContents == null) {
+      throw new IOException("passed Reader cannot be null.");
+    }
+    try (BufferedReader reader = new BufferedReader(fileContents)) {
+      return reader.lines().map(String::trim).collect(Collectors.toList());
+    }
+  }
+
+  /**
+   * A helper function that returns whether file exists.
+   *
+   * @param alternativeMetadataFileName The file name.
+   * @return true if the file exists and is readable and false otherwise.
+   */
+  static boolean fileExistsAndIsReadable(String alternativeMetadataFileName) {
```

Comment:
```suggestion
  static boolean fileExistsAndIsReadable(String filePath) {
```
It's actually the entire file path and the method itself is agnostic of what this is actually about.